peerconnection.go

```diff
@@ -992,6 +992,29 @@ func (pc *PeerConnection) SetRemoteDescription(desc SessionDescription) error {
 			}
 
 			t, localTransceivers = findByMid(midValue, localTransceivers)
+			// Fix simulcast tracks removing
+			if t != nil && t.Receiver() != nil && len(t.Receiver().Tracks()) > 1 {
```

Comment:
This all LGTM! We do need test coverage, I don't think this behavior is Simulcast specific right? We are just reacting to the remotes directions changing?

Reply:
This behaviour is specific to Simulcast since I explained in the commit message this is handled in `PeerConnection.startRTP(l:2010)` but doesn't support Simulcast tracks.

This behaviour can be seen in many renog tests like TestPeerConnection_Renegotiation_AddTrack and TestPeerConnection_Renegotiation_RemoveTrack, but with a simulcast track fails.

Reply:
Can you explain the length check? If it was removed I think this would cover the case from below right?

Reply:
It's to check if current transceiver have a simulcast receiver, If I remove it, it would affect normal tracks, but that is already supported in `PeerConnection.startRTP(l:2010)`

Reply:
It seems like startRTP should be modified to call .Tracks() and loop them instead of .Track() ? https://github.com/pion/webrtc/blob/master/peerconnection.go#L1995. This function will always continue on a transceiver with more than 1 track because .Track() returns nil. It does an == not a >= check https://github.com/pion/webrtc/blob/6e7249351760f23cbc88eaab4f032ee05adad1a8/rtpreceiver.go#L65. Not sure of the implications of any of this @Sean-Der